docs/dev_cycle.intro.rst

```diff
@@ -0,0 +1,134 @@
+e.g. if two nodes suggest two different block for the next block, the nodes gets a decision by consensus.
+
+**Consensus**: Miners get rewards when they find a valid block, and a valid block is valid only if it's built on a valid parent block,
+and *accepted by the majority of nodes on the blockchain*. So miners are incentivised to reject false blocks and false transactions. 
+They know that if they work on a false transaction (say a cheat), then there
+is high probability that other nodes will reject it, and their work effort will be lost without rewards.
+They prefer to find valid blocks with valid transacions, and send them as fast as possible to the blockchain.
+
+**Uncles**: Miners get rewards when they find valid blocks, even if those blocks are *not* part 
+of the direct line of the blockchain. 
+If the blockchain is ``block1`` >> ``block2`` >> ``block3`` >> ``block4``, and a miner found another valid block on top of ``block3``, say ``block4a``,
+but wasn't fast enough to introduce it to the chain, it will still get *partial* rewards.
+The faster miner, of ``block4``, will get the *full* rewards. ``block4`` is included in the direct sequence of the blockchain,
+and ``block4a``  is not used in the sequence but included as an "*uncle*".
+The idea is to spread compenstatations among miners and avoid "the winner takes it all" monopoly.
+
+**Contract**: The word ""contract" is used for three different (albeit related) concepts: 
+(1) A compiled runnable bytecode that sits on the blockchain (2) A Solidity source code contract definition
+(3) A Web3 contract object
+
+**EVM**: The Ethereum Virtual Machine, the (quite complex) piece of code that runs the Ethereum protocol. It accepts an Assembler like instructions,
+and can run contracts after compilation to  this Assembler bytecode.
+
+**Solidity**: A programming language, similar to javascript, designed for contract authors.
+
+**Solc**: Compiler of Solidity source code to the EVM bytecode
+
+**ABI**: Application Binary Interface. A JSON file that describes a contract interface: the functions that the
+contract exposes, and their arguments. Since the contracts on the blockchain are a compiled bytecode,
+the EVM needs the ABI in order to know how to call the bytecode.
+
+**Web3**: Client side API that lets you interact with the blockchain. Web3.js is the javascript version, Web3.py is the Python one.
+
+**geth**: The official implemntation of an Ethereum blockchain node, written in Go
+
+**gas**: The price that users pay to run computational actions on the blockchain (deploying a new contract, send money, run a contract function, storage)
+
+**mainnet**: The Ethereum blockchain
+
+**testnet**: An Ethereum blockchain for testing. It behaves exactly as mainnet, but you don't use real to pay for the Ether and the gas
+
+**Local chain**: A blockchain that runs localy, has it's own blocks, and does not sync to any other blockchain. Useful for development
```

Comment:
All of these might be well suited for a glossary section of the docs that we can the link to when we use the words in tutorials?